### PR TITLE
Fix the Julia set shader example

### PR DESCRIPTION
**Pull request by joshcol9232**
A follow-up to #3466

Hello! Ever since I added the Julia set shader example (~4 years ago!) it had always bugged me that the zoom function was never correct. This PR fixes this, and is also a general tidy-up.

### In summary:

- Fix zoom issue.
- Add a `COLOR_CYCLES` variable to the shader to allow for cycling of the color palette. This allows for finer details of the fractal to be shown when using a high max iteration number.
- Allow user to reset the camera with "R".
- Fix some pretty terrible if statement mess.
- Other minor tidying (removal of unnecessary uniform value).
- Update copyright.

### Issue

Fixes: #3465

### Testing

I have tested the code on my Linux machine. I have not tested on glsl100 platforms, or Windows.

### CODE REVIEW

File: examples/shaders/resources/shaders/glsl100/julia_set.fs [hunk 1]
```diff
@@ -55,7 +56,9 @@ void main()
 
     // The pixel coordinates are scaled so they are on the mandelbrot scale
     // NOTE: fragTexCoord already comes as normalized screen coordinates but offset must be normalized before scaling and zoom
-    vec2 z = vec2((fragTexCoord.x + offset.x/screenDims.x)*2.5/zoom, (fragTexCoord.y + offset.y/screenDims.y)*1.5/zoom);
+    vec2 z = vec2((fragTexCoord.x - 0.5) * 2.5, (fragTexCoord.y - 0.5) * 1.5) / zoom;
```
Comment (raysan5) [inline examples/shaders/resources/shaders/glsl100/julia_set.fs:59]: raylib coding convention: avoid spaces around `*` and `/`

File: examples/shaders/resources/shaders/glsl100/julia_set.fs [hunk 2]
```diff
@@ -79,5 +82,5 @@ void main()
 
     // If in set, color black. 0.999 allows for some float accuracy error.
     if (norm > 0.999) gl_FragColor = vec4(0.0, 0.0, 0.0, 1.0);
-    else gl_FragColor = vec4(Hsv2rgb(vec3(norm, 1.0, 1.0)), 1.0);
+    else gl_FragColor = vec4(Hsv2rgb(vec3(norm * COLOR_CYCLES, 1.0, 1.0)), 1.0);
```
Comment (raysan5) [inline examples/shaders/resources/shaders/glsl100/julia_set.fs:85]: raylib coding convention: avoid spaces around `*` and `/`

File: examples/shaders/shaders_julia_set.c [hunk 3]
```diff
@@ -93,59 +90,63 @@ int main(void)
...
         {
-            if (IsKeyPressed(KEY_ONE)) c[0] = pointsOfInterest[0][0], c[1] = pointsOfInterest[0][1];
-            else if (IsKeyPressed(KEY_TWO)) c[0] = pointsOfInterest[1][0], c[1] = pointsOfInterest[1][1];
-            else if (IsKeyPressed(KEY_THREE)) c[0] = pointsOfInterest[2][0], c[1] = pointsOfInterest[2][1];
-            else if (IsKeyPressed(KEY_FOUR)) c[0] = pointsOfInterest[3][0], c[1] = pointsOfInterest[3][1];
-            else if (IsKeyPressed(KEY_FIVE)) c[0] = pointsOfInterest[4][0], c[1] = pointsOfInterest[4][1];
-            else if (IsKeyPressed(KEY_SIX)) c[0] = pointsOfInterest[5][0], c[1] = pointsOfInterest[5][1];
-
-            SetShaderValue(shader, cLoc, c, SHADER_UNIFORM_VEC2);
+            if (IsKeyPressed(k))
```
Comment (raysan5) [inline examples/shaders/shaders_julia_set.c:95]: Actually for a newby, this code is more complex to understand than previous one... I'd keep the simple version.

File: examples/shaders/shaders_julia_set.c [hunk 4]
```diff
@@ -93,59 +90,63 @@ int main(void)
...
-                offset[0] += GetFrameTime()*offsetSpeed.x*0.8f;
-                offset[1] += GetFrameTime()*offsetSpeed.y*0.8f;
-            }
-            else offsetSpeed = (Vector2){ 0.0f, 0.0f };
+        if (IsKeyPressed(KEY_RIGHT)) incrementSpeed++;
+        else if (IsKeyPressed(KEY_LEFT)) incrementSpeed--;
 
+        // If either left or right button is pressed, zoom in/out.
+        const bool leftMouseButtonDown = IsMouseButtonDown(MOUSE_BUTTON_LEFT);
+        if (leftMouseButtonDown ^ IsMouseButtonDown(MOUSE_BUTTON_RIGHT))
```
Comment (raysan5) [inline examples/shaders/shaders_julia_set.c:123]: Again, the usage of `^` could be really confusing for a newby, even for me. In raylib I always try to keep the code as much clear and simple as possible. Please, could you review it?

**Comment by raysan5** [comment]
@joshcol9232 Thanks for the improvement, added some possible points to review.

File: examples/shaders/shaders_julia_set.c (hunk 3 above)
Comment (joshcol9232) [reply examples/shaders/shaders_julia_set.c]: Thanks, that's fair - reverted in a828267712f514a9a10ad563ede610bf3ee2fb3e

File: examples/shaders/shaders_julia_set.c (hunk 4 above)
Comment (joshcol9232) [reply examples/shaders/shaders_julia_set.c]: I've updated it in [bcc51a2](https://github.com/raysan5/raylib/pull/3467/commits/bcc51a2e7972fe13c0c467ba8140f16023cdd9ac) - is the new version more readable?

**Comment by joshcol9232** [comment]
@raysan5 thank you for the timely review; I've updated given the comments and also tidied elsewhere in the code where the coding conventions were not being followed. Thanks!

**Comment by raysan5** [comment]
@joshcol9232 thanks for the review!

**Pull request merged**